packages/greeter/scripts/exec.js

```diff
@@ -153,35 +153,26 @@ const main = async () => {
     `Greeting txn confirmed on L1! 🙌 ${setGreetingRec.transactionHash}`
   )
 
-  /**
-   * The L1 side is confirmed; now we listen and wait for the for the Sequencer to include the L2 side; we can do this by computing the expected txn hash of the L2 transaction.
-   * To compute this txn hash, we need our message's "sequence number", a unique identifier. We'll fetch from the event logs with a helper method
-   */
-  const inboxSeqNums = await bridge.getInboxSeqNumFromContractTransaction(
-    setGreetingRec
-  )
-  /**
-   * In principle, a single txn can trigger many messages (each with its own sequencer number); in this case, we know our txn triggered only one. Let's get it, and use it to calculate our expected transaction hash.
-   */
-  const ourMessagesSequenceNum = inboxSeqNums[0]
-
-  const retryableTxnHash = await bridge.calculateL2RetryableTransactionHash(
-    ourMessagesSequenceNum
-  )
+  const l1TxReceipt = new L1TransactionReceipt(setGreetingRec)
 
   /**
-   * Now we wait for the Sequencer to include it in its off chain inbox.
-   */
-  console.log(
-    `waiting for L2 tx 🕐... (should take < 10 minutes, current time: ${new Date().toTimeString()}`
-  )
-
-
-
-
-  const retryRec = await l2Provider.waitForTransaction(retryableTxnHash)
+   * In principle, a single L1 txn can trigger any number of L1-to-L2 messages (each with its own sequencer number). 
+   * In this case, we know our txn triggered only one
+   * Here, We check if our L1 to L2 message is redeemed on L2
+  */
+  const message = await l1TxReceipt.getL1ToL2Message(l2Wallet)
+  const status = message.waitForStatus()
+  const isJustEthDeposit = false   // only the user creating the transaction can know whether it was just an eth deposit
```

Comment:
I think you can remove this var, since you know in the script that it's false, and remove the conditional logic below as well to just keep:
```
if(status === L1ToL2TransactionStatus.REDEEMED) {
  console.log(`L2 retryable txn executed 🥳 ${message.l2TxHash}`)
  } else {
console.log(`L2 retryable txn failed with status ${L1ToL2MessageStatus[status]}`)
}
```

Reply:
updated.